### PR TITLE
Persistence configuration changes, k8s secret for db credentials.

### PR DESCRIPTION
**Pull request by ThinkBriK**
Moved sql credentials so that `read` users cannot read the actual secrets.

### CODE REVIEW

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
Hello

1. Plz read contributing doc for the requirements regarding PRs.
2. This PR actually doesn't really solve security issue. Since username / password are still provided via values file, those values can be easily inspected by anyone who has access to k8a cluster, by calling `helm get values`. My suggestion will be to get rid of username and password and instead introduce a new values
* db secret name - a secret containing db creds
* db username fileld - field of db creds secret contining username
* db password field - ...

wdyt ?

**Comment by ThinkBriK** [comment]
> Hello
> 
> 1. Plz read contributing doc for the requirements regarding PRs.
> 2. This PR actually doesn't really solve security issue. Since username / password are still provided via values file, those values can be easily inspected by anyone who has access to k8a cluster, by calling `helm get values`. My suggestion will be to get rid of username and password and instead introduce a new values
> 
> * db secret name - a secret containing db creds
> * db username fileld - field of db creds secret contining username
> * db password field - ...
> 
> wdyt ?

Even better ! BTW helm get value relies on the ability to read releases which are secrets too.

**Comment by ThinkBriK** [comment]
@eshepelyuk : would you mind if I include an explicit choice regarding the persistence? So far the choice relies on the existence of parameters in `values.yaml`.
In my opinion, an explicit choice would be better, just in case you're using some kind of hierarchical values management (using several valuefiles that override the previous ones).

**Comment by eshepelyuk** [comment]
> @eshepelyuk : would you mind if I include an explicit choice regarding the persistence? So far the choice relies on the existence of parameters in `values.yaml`. In my opinion, an explicit choice would be better, just in case you're using some kind of hierarchical values management (using several valuefiles that override the previous ones).

i just dont understand what you mean here :((
so, go make changes I'll gie a feedback.

File: values.schema.json [hunk 1]
```diff
@@ -29,6 +29,10 @@
       }
     },
     "persistence": {
+      "choice": {
```
Comment (eshepelyuk) [inline values.schema.json:32]: A default value mustbe declared.

File: values.schema.json [hunk 2]
```diff
@@ -38,11 +42,12 @@
       },
       "sql": {
         "type": "object", "additionalProperties": false,
-        "required": ["url", "username", "password"],
+        "required": ["url", "secretName", "secretUserField","secretPasswordField"],
         "properties": {
           "url": {"type": "string"},
-          "username": {"type": "string"},
-          "password": {"type": "string"}
+          "secretName": {"type": "string"},
```
Comment (eshepelyuk) [inline values.schema.json:48]: 1. Adjust new fields names
    * `existingSecret`
    * `existingSecretUsernameKey`
    * `existingSecretPasswordKey`
1. Declare default values, corresponding to `values.yaml`

File: values.yaml [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
+     bootstrapServers: kafka1:9092,kafka2:9092
 
   ## uncomment and adjust for sql persistence
-  # sql:
-  #   url: jdbc:postgresql://postgres/apicurio-registry
-  #   username: apicurio
-  #   password: apicurio-password
+  sql:
+   url: jdbc:postgresql://postgres/apicurio-registry
+   secretName: myDBSecret
```
Comment (eshepelyuk) [inline values.yaml:13]: More meaningful secret name required, smth like `apicurio-registry-sync-db`

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
Hello, I liked proposed idea regarding `persistence` parameter.

1. Plz address inline comments.
2. You have to fix helper `apicurio-registry.image` to use new `persistence` flag. Also double check all the templates for the possible adjustments needed after introducing this new parameter.
3. This is a breaking change and it must be reflected via conventional commit message.

And the main thing, please read contributing page carefully, this PR misses mandatory things to be accepted.

File: values.schema.json (hunk 1 above)
Comment (ThinkBriK) [reply values.schema.json:32]: it's now done

File: values.schema.json (hunk 2 above)
Comment (ThinkBriK) [reply values.schema.json]: Done

File: values.yaml (hunk 3 above)
Comment (ThinkBriK) [reply values.yaml]: Done

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
`title` must be added to every new values.yaml property in json schema.

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
Hello, thanks for addressing my minor requests,. although a bigger part is still missing

1. PR jobs are still failing.
2. https://github.com/eshepelyuk/apicurio-registry-helm/blob/main/CONTRIBUTING.md#test-coverage
3. 
Screenshot_2022-12-22-18-55-16-688~2
4. 
Screenshot_2022-12-22-19-02-45-408~2

**Review by eshepelyuk — COMMENTED** [review]
Hello

Is it a state of PR you're considering as a final ? i.e. no more changes expected ?

**Comment by ThinkBriK** [comment]
@eshepelyuk : yes I think it's final

**Comment by ThinkBriK** [comment]
@eshepelyuk : any news ?

**Comment by eshepelyuk** [comment]
Hello

Basically no updates from my side :(

* I can't merge this PR as is, since it' still misses the requested/required changes
* I don't have time atm and in nearest weeks to address those changes myself 

So, unless you're going to address pending issues, ithe PR will be kept open until I get some free time to complete it.

**Comment by eshepelyuk** [comment]
> OK, I think I guessed what you asked. If not, tell me so but it's vague.

I don't feel I understood correctly your reply, but, assuming you're asking about required changes, I'll repeat/rephrase/ask for them once again.

1. **Linter** and **unit** (!!) tests are required, what was provided is zero testing comparing to the size of a change. In fact this change requires e2e test(s) to be added, but I don't even expect this, cause it's quite complex. Linter tests are testing the schema, i.e. every single change to it. Unit tests are testing template rendering depending on values.
2. PR must be squashed to a single commit, the actual feature is change of *persistence configuration*, introduction of DB secret is just a consequence of it.

**Pull request closed without merging**